internal/provider/cluster_resource.go

```diff
@@ -812,6 +915,11 @@ func (r *clusterResource) ModifyPlan(
 				"To prevent accidental deletion of data, changing a cluster's cloud provider "+
 					"isn't allowed. Please explicitly destroy this cluster before changing its cloud provider.")
 		}
+		// BYOC details are immutable after creation.
+		if !byocEqual(plan.CustomerCloudAccount, state.CustomerCloudAccount) {
+			resp.Diagnostics.AddError("Cannot update customer_cloud_account_details",
+				"Changing whether a cluster is hosted in a customer cloud account or modifying its details is not allowed. Destroy and re-create the cluster to change BYOC settings.")
```

Comment:
is BYOC the official marketing acronym for this feature?  We might want to check with docs on that.  @jhlodin

Reply:
Hmm no not really, I'll just change it to say "... to change customer cloud account settings.".

Reply:
Agreed, I removed the BYOC term.